pkg/manager/member/dm_worker_member_manager.go

```diff
@@ -0,0 +1,529 @@
+	if err != nil {
+		return false, fmt.Errorf("workerStatefulSetIsUpgrading: failed to list pods for cluster %s/%s, selector %s, error: %v", dc.GetNamespace(), instanceName, selector, err)
+	}
+	for _, pod := range workerPods {
+		revisionHash, exist := pod.Labels[apps.ControllerRevisionHashLabelKey]
+		if !exist {
+			return false, nil
+		}
+		if revisionHash != dc.Status.Worker.StatefulSet.UpdateRevision {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// syncWorkerConfigMap syncs the configmap of dm-worker
+func (wmm *workerMemberManager) syncWorkerConfigMap(dc *v1alpha1.DMCluster, set *apps.StatefulSet) (*corev1.ConfigMap, error) {
+	if dc.Spec.Worker.Config == nil {
+		return nil, nil
+	}
+	newCm, err := getWorkerConfigMap(dc)
+	if err != nil {
+		return nil, err
+	}
+	return wmm.typedControl.CreateOrUpdateConfigMap(dc, newCm)
+}
+
+func getNewWorkerSetForDMCluster(dc *v1alpha1.DMCluster, cm *corev1.ConfigMap) (*apps.StatefulSet, error) {
+	ns := dc.Namespace
+	dcName := dc.Name
+	baseWorkerSpec := dc.BaseWorkerSpec()
+	instanceName := dc.GetInstanceName()
+	workerConfigMap := cm.Name
+
+	annMount, annVolume := annotationsMountVolume()
+	volMounts := []corev1.VolumeMount{
+		annMount,
+		{Name: "config", ReadOnly: true, MountPath: "/etc/dm-worker"},
+		{Name: "startup-script", ReadOnly: true, MountPath: "/usr/local/bin"},
+		{Name: v1alpha1.DMWorkerMemberType.String(), MountPath: dmWorkerDataVolumeMountPath},
+	}
+	if dc.IsTLSClusterEnabled() {
+		volMounts = append(volMounts, corev1.VolumeMount{
+			Name: "dm-worker-tls", ReadOnly: true, MountPath: "/var/lib/dm-worker-tls",
+		})
+	}
+
+	vols := []corev1.Volume{
+		annVolume,
+		{Name: "config",
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: workerConfigMap,
+					},
+					Items: []corev1.KeyToPath{{Key: "config-file", Path: "dm-worker.toml"}},
+				},
+			},
+		},
+		{Name: "startup-script",
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: workerConfigMap,
+					},
+					Items: []corev1.KeyToPath{{Key: "startup-script", Path: "dm_worker_start_script.sh"}},
+				},
+			},
+		},
+	}
+	if dc.IsTLSClusterEnabled() {
+		vols = append(vols, corev1.Volume{
+			Name: "dm-worker-tls", VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: util.ClusterTLSSecretName(dc.Name, label.DMWorkerLabelVal),
+				},
+			},
+		})
+	}
+
+	storageRequest, err := controller.ParseStorageRequest(dc.Spec.Worker.Requests)
+	if err != nil {
+		return nil, fmt.Errorf("cannot parse storage request for dm-worker, dmcluster %s/%s, error: %v", dc.Namespace, dc.Name, err)
+	}
+
+	workerLabel := label.NewDM().Instance(instanceName).DMWorker()
+	setName := controller.DMWorkerPeerMemberName(dcName)
```

Comment:
I prefer use `demo-dm-worker` instead of `demo-dm-worker-peer` for DMWorker's statefulset name.

Reply:
+1